Use Turbo Stream to update comments list on deletion

gerardoacuna opened the pull request:
When a comment is destroyed, replace the entire comments section with the updated list partial instead of just removing the DOM element. This ensures the UI properly reflects the current state and handles edge cases like empty state rendering.

Fixes #11601

Luke-Oldenburg commented on app/views/comments/_list.html.erb:
```diff
@@ -1,8 +1,7 @@
-<section class="comments">
```
Are we losing styling by switching this from class to id or was it a typo?

gerardoacuna replied:
The ID is needed because it's referenced by the controller and I didn't notice any style changes. I'll leave both just to be sure.

gerardoacuna committed:
Use Turbo Stream to update comments list on deletion

When a comment is destroyed, replace the entire comments section with
the updated list partial instead of just removing the DOM element.
This ensures the UI properly reflects the current state and handles
edge cases like empty state rendering.

Fixes #11601

Luke-Oldenburg reviewed (approved):
lgtm!

Luke-Oldenburg committed:
Merge branch 'main' into fix/turbo-stream-comments-list

Luke-Oldenburg commented on app/controllers/comments_controller.rb:
```diff
@@ -48,14 +48,21 @@ def update
+          partial: "comments/list",
+          locals: {
+            comments: commentable.comments,
+            show_blankslate: commentable.comments.empty? }
```
```suggestion
            show_blankslate: commentable.comments.empty?
          }
```

Luke-Oldenburg committed:
Apply suggestions from code review
